molgenis-data-import/src/main/java/org/molgenis/data/importer/emx/EmxMetadataParser.java

```diff
@@ -526,16 +526,18 @@ IntermediateParseResults parseSinglePackage(
     return intermediateResults;
   }
 
-  /** Convert tag identifiers to tags */
-  private static List<Tag> toTags(
-      IntermediateParseResults intermediateResults, List<String> tagIdentifiers) {
+  /** Convert tag identifiers to tags Package private for unit test */
+  List<Tag> toTags(IntermediateParseResults intermediateResults, List<String> tagIdentifiers) {
     if (tagIdentifiers.isEmpty()) {
       return emptyList();
     }
 
     List<Tag> tags = new ArrayList<>(tagIdentifiers.size());
     for (String tagIdentifier : tagIdentifiers) {
-      Tag tag = intermediateResults.getTag(tagIdentifier);
+      Tag tag =
+          intermediateResults.hasTag(tagIdentifier)
+              ? intermediateResults.getTag(tagIdentifier)
+              : dataService.findOneById(TAG, tagIdentifier, Tag.class);
```

Comment:
Retrieving tags one-by-one is a potential performance issue. Consider determining which tags need to be retrieved via the data service and retrieve with a findAll.